ParquetClassLibrary/Sandbox/RoomRecipe.cs

```diff
@@ -0,0 +1,115 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace ParquetClassLibrary.Sandbox
+{
+    /// <summary>
+    /// Models the minimum requirements for a <see cref="Room"/> to be recognizable and useful.
+    /// </summary>
+    public class RoomRecipe : Entity
+    {
+        #region Recipe Requirements
+        /// <summary>
+        /// Minimum number of open spaces needed for this <see cref="RoomRecipe"/> to register.
+        /// </summary>
+        public int MinimumWalkableSpaces { get; }
+
+        /// <summary>An optional list of <see cref="Sandbox.Parquets.Floor"/> types this <see cref="RoomRecipe"/> requires.</summary>
+        public List<EntityID> RequiredFloors { get; }
+
+        /// <summary>An optional list of <see cref="Sandbox.Parquets.Block"/> types this <see cref="RoomRecipe"/> requires as walls.</summary>
+        public List<EntityID> RequiredPerimeterBlocks { get; }
+
+        /// <summary>A list of <see cref="Sandbox.Parquets.Furnishing"/> types this <see cref="RoomRecipe"/> requires.</summary>
+        public Dictionary<EntityID, int> RequiredFurnishings { get; }
```

Comment:
question: This is a dictionary with a countable amounts rather than a pure list. Is this to suggest that a room might require 3 of a furnishing before it counts?

Reply:
Yes, exactly.
Like, a bedroom might require a bed, but a barracks would require at least 12 beds.